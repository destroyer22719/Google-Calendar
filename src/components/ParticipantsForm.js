import React, { useState } from "react";
import ParticipantItem from "./ParticipantItem";

const ParticipantsForm = ({state, dispatch}) => {
    const [newParticipant, setNewParticipant] = useState("");

    const submitParticipant = e => {
        e.preventDefault();
        dispatch({
            type: "ADD_PARTICIPANT",
            participant: newParticipant,
        });
        setNewParticipant("");
    }

    return (
        <div>
            <form onSubmit={submitParticipant}>
                <label htmlFor="participant">Add Participant</label>
                <input type="email" id="participant" value={newParticipant} onChange={e => setNewParticipant(e.target.value)}/>
                <button>Add Participant</button>
            </form>
            {state.map(({email: participant}) => <ParticipantItem key={participant} email={participant} dispatch={dispatch}/>)}
        </div>
    );
};

export default ParticipantsForm;