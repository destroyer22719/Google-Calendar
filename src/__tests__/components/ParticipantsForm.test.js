import React from "react";
import {shallow} from "enzyme";
import ParticipantsForm from "../../components/ParticipantsForm";

let wrapper, setNewParticipant;

beforeEach(() => {
    setNewParticipant = jest.fn();

    wrapper = shallow(
        <ParticipantsForm 
            state={[]}
            setNewParticipant={setNewParticipant}
        />
    );
});

test("Should render ParticipantsForm component", () => {
    expect(wrapper).toMatchSnapshot();
})