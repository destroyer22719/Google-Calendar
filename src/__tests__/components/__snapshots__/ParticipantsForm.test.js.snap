// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ParticipantsForm component 1`] = `
<div>
  <form
    onSubmit={[MockFunction]}
  >
    <label
      htmlFor="participant"
    >
      Add Participant
    </label>
    <input
      id="participant"
      onChange={[Function]}
      type="email"
      value=""
    />
    <button
      id="addParticipant"
    >
      Add Participant
    </button>
  </form>
</div>
`;
